cluster-controller/src/test/java/io/strimzi/controller/cluster/operations/kubernetes/CreateOperationTest.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.strimzi.controller.cluster.operations.kubernetes;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.api.model.ConfigMapBuilder;
+import io.strimzi.controller.cluster.K8SUtils;
+import io.strimzi.controller.cluster.MockK8sUtils;
+import io.strimzi.controller.cluster.operations.CreateOperation;
+import io.vertx.core.Vertx;
+import io.vertx.ext.unit.Async;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.Collections;
+
+import static java.util.Collections.singletonMap;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(VertxUnitRunner.class)
```

Comment:
I think that technically this test is good. However I'm not completely sure about the added value. Because right now the `CreateOperation` is just a proxy to `K8SUtils` / `OpenShiftUtils`. This PR has a big added value in that it simplifies the error handling from being a duplicated across different `Create*` files. These ConfigMap tests now test the login in the `execute` method, which is good, although I think it is low risk area when put into single place. But writing further tests for the other Kubernetes / OpenShift components would be without any significant added value.

It relates a bit to the point I raised above about what to do with `K8SUtils` / `OpenShiftUtils`. We have two layers of abstraction for the same thing. And it is not clear to me whether that has still some meaning.

Reply:
As it stands, I wouldn't write any more tests of the `CreateOperation` (to cover other resource types), since I think this is basically enough. I do think is has some value though, as it allows us to test that we always call the handler even in error cases. 

If we end up removing the `*Utils` classes then there would be a little more value in having per-create operation tests, because each operation would be doing something different with the fabric8 API. In that case it might be possible to refactor the test too, so that testing all the create operations isn't too difficult, but if that wasn't possible I would probably still just test one case, on the basis that an integration-style test should catch problems in how we use the fabric8 API.

Reply:
@scholzj I agree with you about these two abstraction layers we have but as far as I can see we still use the `K8SUtils` in some places which are not operations related (i.e. the `isOpenShift` in the various `diff`, `fromConfigMap` and so on). Btw I agree that if possible we could reduce to have only one abstraction layer.